### ISSUE TITLE
`SAVE ARTIFACT` not updating local file with `AS LOCAL` after `RUN --no-cache`

### ISSUE THREAD

**Issue by dchw** [earthly/earthly#1695]
Given the following target:

```
test:
    FROM bash

    RUN --no-cache true
    COPY incr .
    RUN /bin/ash -c "echo $(($(cat incr) + 1)) >incr"

    SAVE ARTIFACT --force incr AS LOCAL incr
```

Perform the following steps:
1. Make a file named `incr` with a single number in it to start (I used 1)
2. Run this, see that `incr` went to the next number (2)
3. Run it again, see that `incr` went to the next number (3)
4. Externally edit `incr` to an earlier number (back to 2)
5. Re-run the target. see that `incr` went to the next number (3)
6. Re-run the target. see that `incr`  went to the next number (4)
7. Re-run the target, see that `incr` did not go to the next number.

This also works (and gets stuck at 3) if you go back to 1.

In conversation with @alexcb, we discovered that this is because the file size, and modification time of the new and old files matched, despite the differing contents. So, we worked around this by using `SAVE ARTIFACT --keep-ts` - which means that the last modification time will always cause us to re-write the file, even if we go back in sequence.

**Comment by adamgordonbell** [comment]
I think I got hit by this in cert generation code here:
https://github.com/adamgordonbell/cloudservices/blob/gateway/activity-log/Earthfile#L44

**Comment by vladaionescu** [comment]
It seems that the following buildkit change is the cause of this: https://github.com/moby/buildkit/pull/2480

> In situations where the user actually did change a file and it shows up
> in the upperdir, it should be extremely rare that the content comparison
> code path is followed. It would require that the user changed no other
> metadata of the file, including size, and both mod timestamps were the
> same (which could only really happen if their underlying filesystem
> lacked support for nanosecond precision and they modified the file
> within 1 second of its modification in the lowerdir or they manually
> changed the modtime with chtimes).

A quick fix could be to disable the new differ in Earthly for now.

**Comment by vladaionescu** [comment]
> It seems that the following buildkit change might be related to this: https://github.com/moby/buildkit/pull/2480

That ended up being a red herring.

It was something different: when outputting locally, we were outputting with a flattened timestamp - and when loading files into Earthly, we loaded them by looking at the timestamp (among other things) to figure out if the file had changed. To fix this, I've made a change to update the timestamp to files when we output them locally - this fixes the issue.
